### PR TITLE
TEIIDTOOLS-986 adding intial service methods for table privileges

### PR DESCRIPTION
**Pull request by shawkins**

### CODE REVIEW

**Comment by shawkins** [comment]
I need to run this through more end-to-end testing, but the minimal set of role backend logic is there for the ui.

File: app/dv/src/main/java/io/syndesis/dv/server/endpoint/DataVirtualizationService.java [hunk 1]
```diff
@@ -1065,38 +1100,103 @@ public PodMetrics getPublishedVirtualizationMetrics(
...
+                    metrics.setResultSetCacheHitRatio(hitRatio);
+                }
             }
         }
-        }
 
         return metrics;
     }
 
+    @GetMapping(value = {
```
Comment (rareddy) [inline app/dv/src/main/java/io/syndesis/dv/server/endpoint/DataVirtualizationService.java:1137]: So there is no API for to get View level for getting a role? Maybe it is already on some getView?

File: app/dv/src/main/java/io/syndesis/dv/openshift/TeiidOpenShiftClient.java [hunk 2]
```diff
@@ -1263,6 +1269,15 @@ public BuildStatus publishVirtualization(PublishConfiguration publishConfig) thr
         return properties;
     }
 
+    Collection<EnvVar> getEnvironmentVariablesForSecurity(SSOConfigurationProperties props) {
+        List<EnvVar> envs = new ArrayList<>();
+        envs.add(env("KEYCLOAK_AUTHSERVERURL", props.getAuthServerUrl()));
+        envs.add(env("KEYCLOAK_REALM", props.getRealm()));
+        envs.add(env("KEYCLOAK_RESOURCE", props.getResource()));
```
[the remark this answers is not in the document]
Comment (shawkins) [reply app/dv/src/main/java/io/syndesis/dv/openshift/TeiidOpenShiftClient.java:1276]: Doesn't that require creating a keycloak client per vdb?

File: app/dv/src/main/java/io/syndesis/dv/openshift/TeiidOpenShiftClient.java [hunk 3]
```diff
@@ -1263,6 +1269,15 @@ public BuildStatus publishVirtualization(PublishConfiguration publishConfig) thr
...
+        envs.add(env("KEYCLOAK_SSL_REQUIRED", props.getSslRequired()));
```
[the remark this answers is not in the document]
Comment (shawkins) [reply app/dv/src/main/java/io/syndesis/dv/openshift/TeiidOpenShiftClient.java:1277]: Sure.  We can also carry forward the approach of envFrom by adding all sso prefixed properties as keycloak properties and not use this explicit approach.

File: app/dv/src/main/java/io/syndesis/dv/server/endpoint/DataVirtualizationService.java (hunk 1 above)
Comment (shawkins) [reply app/dv/src/main/java/io/syndesis/dv/server/endpoint/DataVirtualizationService.java:1137]: No, based upon all of our conversations that was not needed yet.

File: app/dv/src/main/java/io/syndesis/dv/openshift/TeiidOpenShiftClient.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rareddy) [reply app/dv/src/main/java/io/syndesis/dv/openshift/TeiidOpenShiftClient.java:1276]: Yes, here that moel does not work until we move to the Operator based deployment where we can probably manage much better as KEYCLOAK configuration moves to that.

[the remark this answers is not in the document]
Comment (shawkins) [reply app/dv/src/main/java/io/syndesis/dv/openshift/TeiidOpenShiftClient.java]: All sso_ properties are now passed.

**Pull request merged**